app/src/main/java/com/sschuraytz/charcoaldrawing/MainActivity.java

```diff
@@ -131,4 +158,121 @@ public void onUndoAvailable(boolean isAvailable) {
     public void onRedoAvailable(boolean isAvailable) {
         updateVisibility(isAvailable, redoButton);
     }
+
+    @RequiresApi(api = Build.VERSION_CODES.M)
+    public void checkVoicePermissions () {
+        if (ContextCompat.checkSelfPermission(
+                this, Manifest.permission.RECORD_AUDIO) == PackageManager.PERMISSION_GRANTED) {
+            //perform();
+        } else {
+           requestPermissions(
+               new String[] { Manifest.permission.RECORD_AUDIO},
+                1
+           );
+        }
+    }
+
+    public void setUpSpeechRecognizer() {
+        speechRecognizer = SpeechRecognizer.createSpeechRecognizer(this);
+        speechRecognizer.setRecognitionListener(recognitionListener);
+    }
+
+    public void setUpRecognitionListener() {
+        recognitionListener = new RecognitionListener() {
+            @Override
+            public void onReadyForSpeech(Bundle params) {
+                Toast.makeText(getApplicationContext(), "listening", Toast.LENGTH_LONG).show();
+            }
+
+            @Override
+            public void onBeginningOfSpeech() {
+            }
+
+            @Override
+            public void onRmsChanged(float rmsdB) {
+
+            }
+
+            @Override
+            public void onBufferReceived(byte[] buffer) {
+
+            }
+
+            @Override
+            public void onEndOfSpeech() {
+                fab.setImageResource(R.drawable.ic_mic_foreground);
+            }
+
+            @Override
+            public void onError(int error) {
+                fab.setImageResource(R.drawable.ic_mic_foreground);
+            }
+
+            //Is there a way to make it more flexible so if user doesn't say exact word it still works?
```

Comment:
What you can do is define multiple words that are similar, like "erase" and "eraser" that do the same thing.